### PR TITLE
Improved LOOCV ridge regression using singular value decomposition

### PR DESCRIPTION
**Pull request by narutak**
I have improved LOOCV ridge regression to be executable with low memory usage by utilizing singular value decomposition.
I made modifications mainly to two files:
- src/symfc_alm/ridge.py: I have fixed the estimation of regression coefficients (RidgeRegression._fit()) and the calculation of LOOCV error (RidgeRegression._calc_error()).
- tests/test_ridge.py: I have added tests.

Please feel free to review the changes and leave any feedback or comments. I look forward to your review of the pull request.

### CODE REVIEW

File: src/symfc_alm/ridge.py [hunk 1]
```diff
@@ -170,14 +174,26 @@ def run_auto(
...
     def _fit(self, alpha: float, debias: bool):
-        """Fit force constants.
+        r"""Fit force constants.
+
+        The regression coefficients for Ridge regression are calculated
+        using singular value decomposition as follows:
+
+        V(R^{T}R + \alpha I)^{-1}R^{T}b
+
+        where R is R = U @ Σ, and U, Σ, V are obtained from the
```
Comment (atztogo) [inline src/symfc_alm/ridge.py]: Please avoid using Greek letter character.

File: src/symfc_alm/ridge.py [hunk 2]
```diff
@@ -238,19 +261,43 @@ def _calc_error(self, alpha: float):
...
         return error
 
+    def _calc_RTR_inv(self, alpha: float):
+        r"""Return the inverse of a specific matrix.
+
+        The inverse is calculated for the following equation:
+
+        (R^{T}R + \alpha I)^{-1}
+
+        where R is R = U @ Σ, and U, Σ are obtained from the
```
Comment (atztogo) [inline src/symfc_alm/ridge.py]: Please avoid using Greek letter character.

**Comment by atztogo** [comment]
I will remove `.vscode/settings.json` on my side.

**Review by atztogo — CHANGES_REQUESTED** [review]

File: src/symfc_alm/ridge.py (hunk 1 above)
Comment (narutak) [reply src/symfc_alm/ridge.py]: Thank you very much. I fixed it.

File: src/symfc_alm/ridge.py (hunk 2 above)
Comment (narutak) [reply src/symfc_alm/ridge.py]: Thank you very much. I fixed it.

**Comment by narutak** [comment]
Please review the code and if there are no issues, I request you to merge it.

**Review by atztogo — APPROVED** [review]

**Comment by atztogo** [comment]
Merged into develop. Thanks @narutak.

**Pull request closed without merging**